common/persistence/visibility/store/elasticsearch/processor.go

```diff
@@ -172,6 +172,11 @@ func (p *processorImpl) hashFn(key interface{}) uint32 {
 
 // Add request to the bulk and return a future object which will receive ack signal when request is processed.
 func (p *processorImpl) Add(request *client.BulkableRequest, visibilityTaskKey string) *future.FutureImpl[bool] {
+	if p.mapToAckFuture == nil { // mapToAckFuture will be nil iff processor has been shut down
+		p.logger.Warn("Skipping ES request for visibility task key because processor has been shut down.", tag.Key(visibilityTaskKey), tag.ESDocID(request.ID), tag.Value(request.Doc))
+		return nil
+	}
```

Comment:
wouldn't there be race between Stop() and Add()?

Reply:
Yeah I have synced offline with @pdoerner on this. Essentially need to drain all pending requests and also block new requests from coming in.